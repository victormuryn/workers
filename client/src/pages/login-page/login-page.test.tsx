import React from 'react';
import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import LoginPage from './login-page';

describe(`LoginPage tests`, () => {
  it(`LoginPage renders corrects`, () => {
    const tree = renderer.create(<Router><LoginPage /></Router>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
