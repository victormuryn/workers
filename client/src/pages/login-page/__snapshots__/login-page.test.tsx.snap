// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage tests LoginPage renders corrects 1`] = `
Array [
  <header
    className="auth-header"
  >
    <h2
      className="auth-header__logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Workers
      </a>
    </h2>
    <p
      className="auth-header__text"
    >
      Увійдіть та почніть змінювати своє життя
    </p>
  </header>,
  <div
    className="container"
  >
    <form
      className="auth-form mt-5"
      method="POST"
      onSubmit={[Function]}
    >
      <label
        className="auth-form__label row"
      >
        <span
          className="auth-form__label-text mb-2 col-md-4 offset-md-2"
        >
          Логін:
        </span>
        <div
          className="position-relative mb-2 col-lg-3 col-md-6"
        >
          <div
            className="auth-page__input-underlined"
          >
            <input
              name="username"
              onChange={[Function]}
              pattern="^(?!.*\\\\.\\\\.)(?!.*\\\\.$)[^\\\\W][\\\\w.]{0,29}$"
              required={true}
              type="text"
              value=""
            />
            <span
              className="auth-page__input-label"
            >
              Ваш логін
            </span>
          </div>
        </div>
      </label>
      <label
        className="auth-form__label row"
      >
        <span
          className="auth-form__label-text mb-2 col-md-4 offset-md-2"
        >
          Пароль:
        </span>
        <div
          className="position-relative mb-2 col-lg-3 col-md-6"
        >
          <div
            className="auth-page__input-underlined"
          >
            <input
              minLength={6}
              name="password"
              onChange={[Function]}
              required={true}
              type="password"
              value=""
            />
            <span
              className="auth-page__input-label"
            >
              Ваш пароль
            </span>
          </div>
        </div>
      </label>
      <label
        className="auth-form__label row"
      >
        <span
          className="text-right col-md-2 offset-md-4"
        />
        <div
          className="position-relative col-md-3"
        >
          <div
            className="auth-page__input-group auth-page__input-group--rules"
          >
            <button
              className="auth-page__submit btn btn-success"
              disabled={false}
              type="submit"
            >
              Увійти
            </button>
          </div>
        </div>
      </label>
      <p
        className="auth-page__login"
      >
        Ще не зареєстровані? 
        <a
          href="/auth"
          onClick={[Function]}
        >
          Зареєструватися
        </a>
      </p>
    </form>
  </div>,
  <footer
    className="footer container-fluid"
    style={
      Object {
        "background": "linear-gradient(
      179deg,
      transparent 0,
      transparent 45px,
      #fff 45px,
      #fff 100%
    )",
      }
    }
  >
    <div
      className="footer__wrapper row row-cols-md-3 row-cols-1"
    >
      <div
        className="footer__column col-md-12 order-md-last"
      >
        <h2
          className="footer__logo"
        >
          <a
            href="/"
          >
            Workers
          </a>
        </h2>
        <p
          className="footer__copy"
        >
          © 2021 workers.com.ua — всі права захищені
        </p>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Реклама на сайті
        </a>
        <a
          href="#"
        >
          Зв'язатися з адміністрацією
        </a>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Угода користувача
        </a>
        <a
          href="#"
        >
          Допомога
        </a>
      </div>
      <div
        className="footer__column col"
      >
        <a
          href="#"
        >
          Замовники
        </a>
        <a
          href="#"
        >
          Виконавці
        </a>
        <a
          href="#"
        >
          Блог
        </a>
        <a
          href="#"
        >
          Безпечна угода
        </a>
      </div>
    </div>
  </footer>,
]
`;
